### PR TITLE
OBSDOCS-900: expand and revise monitoring API access section

### PR DESCRIPTION
**Pull request by bburt-rh**
Version(s): 4.12+

Issue: https://issues.redhat.com/browse/OBSDOCS-900

Link to docs preview: https://72877--ocpdocs-pr.netlify.app/openshift-enterprise/latest/monitoring/accessing-third-party-monitoring-apis

QE review:
- [x] QE has approved this change.

Additional information:
This PR makes following changes meant to improve the content in the [existing API access section of the monitoring docs](https://docs.openshift.com/container-platform/4.15/monitoring/accessing-third-party-monitoring-apis.html):

- include the existing procedure module located at https://docs.openshift.com/container-platform/4.15/monitoring/enabling-monitoring-for-user-defined-projects.html#accessing-metrics-from-outside-cluster_enabling-monitoring-for-user-defined-projects
- revise the [existing API access procedure](https://docs.openshift.com/container-platform/4.15/monitoring/accessing-third-party-monitoring-apis.html#accessing-third-party-monitoring-web-service-apis_accessing-third-party-monitoring-apis) to be consistent with the accessing metrics from outside the cluster topic and to align the formatting with the guidance from the OCP docs guidelines.
- add an `Important` admonition to the beginning of the section in the assembly adoc file informing users of limitations and other issues that might arise when accessing monitoring APIs.

### CODE REVIEW

File: modules/accessing-metrics-outside-cluster.adoc [hunk 1]
```diff
@@ -6,7 +6,7 @@
 [id="accessing-metrics-from-outside-cluster_{context}"]
 = Accessing metrics from outside the cluster for custom applications
 
-Learn how to query Prometheus statistics from the command line when monitoring your own services. You can access monitoring data from outside the cluster with the `thanos-querier` route.
+You can query Prometheus metrics from outside the cluster when monitoring your own services with user-defined projects. You access this data from outside the cluster by using the `thanos-querier` route.
```
Comment (juzhao) [inline modules/accessing-metrics-outside-cluster.adoc:9]:
```suggestion
You can query Prometheus metrics from outside the cluster when monitoring your own services with user-defined projects. Access this data from outside the cluster by using the `thanos-querier` route.
```
maybe it's better

File: modules/monitoring-about-accessing-monitoring-web-service-apis.adoc [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+:_mod-docs-content-type: CONCEPT
+[id="about-accessing-monitoring-web-service-apis_{context}"]
+= About accessing monitoring web service APIs
+
+You can directly access web service API endpoints from the command line for the following monitoring stack components: Prometheus, Alertmanager, Thanos Ruler, and Thanos Querier.
+
+[NOTE]
+====
+To access Thanos Ruler and Thanos Querier service APIs, the requesting account must have `get` permission on the namespaces resource, which can granted by binding the `cluster-monitoring-view` cluster role to the account.
```
Comment (juzhao) [inline modules/monitoring-about-accessing-monitoring-web-service-apis.adoc:13]:
```suggestion
To access Thanos Ruler and Thanos Querier service APIs, the requesting account must have `get` permission on the namespaces resource, which can be granted by binding the `cluster-monitoring-view` cluster role to the account.
```

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc [hunk 3]
```diff
@@ -3,37 +3,42 @@
...
-$ host=$(oc -n openshift-monitoring get route alertmanager-main -ojsonpath={.spec.host})
+$ TOKEN=$(oc whoami -t)
 ----
 
+. Extract the `alertmanager-main` API route URL by running the following command:
++
 [source,terminal]
 ----
-$ token=$(oc whoami -t)
+$ host=$(oc -n openshift-monitoring get route alertmanager-main -ojsonpath={.spec.host})
```
Comment (juzhao) [inline modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:36]:
```suggestion
$ HOST=$(oc -n openshift-monitoring get route alertmanager-main -ojsonpath={.spec.host})
```

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc [hunk 4]
```diff
@@ -3,37 +3,42 @@
...
 ----
 
+. Query the service API receivers for Alertmanager by running the following command:
++
 [source,terminal]
 ----
 $ curl -H "Authorization: Bearer $token" -k "https://$host/api/v2/receivers"
```
Comment (juzhao) [inline modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:43]:
```suggestion
$ curl -H "Authorization: Bearer $TOKEN" -k "https://$HOST/api/v2/receivers"
```

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc (hunk 3 above)
Comment (juzhao) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:36]: and 
https://72877--ocpdocs-pr.netlify.app/openshift-enterprise/latest/monitoring/accessing-third-party-monitoring-apis#monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus_accessing-third-party-monitoring-apis
```
$ token=`oc whoami -t`
```
change to below is better
```
$ TOKEN=`oc whoami -t`
```

File: modules/monitoring-about-accessing-monitoring-web-service-apis.adoc (hunk 2 above)
Comment (bburt-rh) [reply modules/monitoring-about-accessing-monitoring-web-service-apis.adoc:13]: fixed

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc (hunk 3 above)
Comment (bburt-rh) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:36]: Fixed

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc (hunk 4 above)
Comment (bburt-rh) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:43]: fixed

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc [hunk 5]
```diff
@@ -3,37 +3,42 @@
...
-This example requires that the associated user account be bound against the `monitoring-alertmanager-edit` role in the `openshift-monitoring` namespace and that the account has the privilege to view the route.
-This access only supports using a Bearer Token for authentication.
+.Prerequisites
 
-[source,terminal]
-----
-$ oc login -u <username> -p <password>
-----
+* You are logged in to an account that is bound against the `monitoring-alertmanager-edit` role in the `openshift-monitoring` namespace.
+* You are logged in to an account with the `cluster-monitoring-view` cluster role, which provides permission to access the Alertmanager API.
```
[the remark this answers is not in the document]
Comment (bburt-rh) [reply modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:15]: Removed this item from the list.

File: modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc [hunk 6]
```diff
@@ -3,37 +3,41 @@
...
-$ token=$(oc whoami -t)
+$ HOST=$(oc -n openshift-monitoring get route alertmanager-main -ojsonpath={.spec.host})
 ----
 
+. Query the service API receivers for Alertmanager by running the following command:
++
 [source,terminal]
 ----
-$ curl -H "Authorization: Bearer $token" -k "https://$host/api/v2/receivers"
+$ curl -H "Authorization: Bearer $token" -k "https://$HOST/api/v2/receivers"
```
Comment (juzhao) [inline modules/monitoring-accessing-third-party-monitoring-web-service-apis.adoc:42]:
```suggestion
$ curl -H "Authorization: Bearer $TOKEN" -k "https://$HOST/api/v2/receivers"
```

**Comment by juzhao** [comment]
https://72877--ocpdocs-pr.netlify.app/openshift-enterprise/latest/monitoring/accessing-third-party-monitoring-apis#monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus_accessing-third-party-monitoring-apis
since
```
$ token=`oc whoami -t`
```
is changed to
```
$ TOKEN=`oc whoami -t`
```
then we should also update for
```
curl -G -s -k -H "Authorization: Bearer $token" 'https://<federation_host>/federate' --data-urlencode 'match[]=up'
```

File: modules/monitoring-about-accessing-monitoring-web-service-apis.adoc [hunk 7]
```diff
@@ -0,0 +1,16 @@
+// Module included in the following assemblies:
+//
+// * monitoring/accessing-third-party-monitoring-uis-and-apis.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="about-accessing-monitoring-web-service-apis_{context}"]
+= About accessing monitoring web service APIs
+
+You can directly access web service API endpoints from the command line for the following monitoring stack components: Prometheus, Alertmanager, Thanos Ruler, and Thanos Querier.
```
Comment (xenolinux) [inline modules/monitoring-about-accessing-monitoring-web-service-apis.adoc:9]: I think an unnumbered bullet list can be used here.

**Review by xenolinux — COMMENTED** [review]
One thing - up to you to consider; otherwise LGTM

/remove-label peer-review-needed
/remove-label peer-review-in-progress
/label peer-review-done

File: modules/monitoring-about-accessing-monitoring-web-service-apis.adoc (hunk 7 above)
Comment (bburt-rh) [reply modules/monitoring-about-accessing-monitoring-web-service-apis.adoc]: Agree. Changed.

[the remark this answers is not in the document]
Comment (bburt-rh) [reply modules/monitoring-about-accessing-monitoring-web-service-apis.adoc]: changed per your suggestion

**Pull request merged**